SQRLPlatformAwareInstaller/ViewModels/VersionSelectorViewModel.cs

```diff
@@ -360,5 +360,81 @@ public void Cancel()
             ((MainWindowViewModel)AvaloniaLocator.Current.GetService<MainWindow>().DataContext).Content = ((MainWindowViewModel)AvaloniaLocator.Current.GetService<MainWindow>().DataContext).PriorContent;
         }
 
+        public void ExtractZipFile(string archivePath, string password, string outFolder)
+        {
+
+            using (Stream fsInput = File.OpenRead(archivePath))
+            using (var zf = new ICSharpCode.SharpZipLib.Zip.ZipFile(fsInput))
+            {
+                if (!String.IsNullOrEmpty(password))
+                {
+                    // AES encrypted entries are handled automatically
+                    zf.Password = password;
+                }
+                long fileCt = zf.Count;
+                int ct = 0;
+                foreach (ZipEntry zipEntry in zf)
+                {
+                    ct++;
+                    this.DownloadPercentage = (int)((ct / fileCt) * 20M);
+                    if (!zipEntry.IsFile)
+                    {
+                        // Ignore directories
+                        continue;
+                    }
+                    String entryFileName = zipEntry.Name;
+                    // to remove the folder from the entry:
+                    //entryFileName = Path.GetFileName(entryFileName);
+                    // Optionally match entrynames against a selection list here
+                    // to skip as desired.
+                    // The unpacked length is available in the zipEntry.Size property.
+
+                    // Manipulate the output filename here as desired.
+                    var fullZipToPath = Path.Combine(outFolder, entryFileName);
+                    if(entryFileName.Equals("sqrl.db", StringComparison.OrdinalIgnoreCase) && File.Exists(entryFileName) )
```

Comment:
We should probably replace the hardcoded db name with a static readonly variable sometime in the future, not only here but everywhere else as well. Doesn't need to be now though.